src/App.js

```diff
@@ -0,0 +1,36 @@
+import React, { useState, useEffect } from 'react';
+import AddTodoForm from './AddTodoForm';
+import TodoList from './TodoList';
+
+const useSemiPersistentState = () => {
+  const [todoList, setTodoList] = useState(JSON.parse(localStorage.getItem("savedTodoList")) || []);
+
```

Comment:
Just as an aside, if you ever find your todo list app malfunctioning, open up the application tab on your browser (Chrome, Brave, Edge, etc) and you can find your browser's local storage for the localhost site. You can delete that key-value pair and it will wipe out the storage, letting you start fresh.